# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsContentUtils.cpp
# Commit: 420e02d37505
# Full Hash: 420e02d3750587b6dfaa72229a38e4ff8bb38149
# Author: Tom Schuster <evilpies@gmail.com>
# Date: 2022-10-19 04:09:55
# Regressor Bug: 1778565
# File Overlap Count: 1
# Description:
#   Bug 1778565 - Editor: Handle nsIFile transferable as blob. r=nika,masayuki
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D155753
# ==============================================================================

diff -r 3f19cab95424 -r 420e02d37505 dom/base/nsContentUtils.cpp
--- a/dom/base/nsContentUtils.cpp	Tue Oct 18 17:35:29 2022 +0000
+++ b/dom/base/nsContentUtils.cpp	Tue Oct 18 17:35:33 2022 +0000
@@ -8086,28 +8086,6 @@
       // Otherwise, handle this as a file.
       nsCOMPtr<BlobImpl> blobImpl;
       if (nsCOMPtr<nsIFile> file = do_QueryInterface(data)) {
-        // FIXME(bug 1778565): Historically, attempting to send a Blob over IPC
-        // in response to a sync message would lead to a crash, however this
-        // isn't an issue anymore. Unfortunately, code in HTMLEditor depends on
-        // the old behaviour, so we need to preserve this oddity, and hope that
-        // the caller is HTMLEditor and can handle a nsIInputStream.
-        if (aInSyncMessage) {
-          nsAutoCString type;
-          if (IsFileImage(file, type)) {
-            IPCDataTransferItem* item =
-                aIPCDataTransfer->items().AppendElement();
-            item->flavor() = type;
-            nsCString data;
-            SlurpFileToString(file, data);
-            // FIXME: This can probably be simplified once bug 1783240 lands, as
-            // `nsCString` will be implicitly serialized in shmem when sent over
-            // IPDL directly.
-            item->data() =
-                IPCDataTransferInputStream(BigBuffer(AsBytes(Span(data))));
-          }
-          continue;
-        }
-
         if (aParent) {
           bool isDir = false;
           if (NS_SUCCEEDED(file->IsDirectory(&isDir)) && isDir) {